pyramid_swagger/ingest.py

```diff
@@ -21,93 +21,80 @@ class ApiDeclarationNotFoundError(Exception):
 
     :param schema_dir: the directory schema files live inside
     :type schema_dir: string
-    :returns: A tuple of (resource listing filepath, mapping) where the mapping
-        is between resource name and file path
-    :rtype: (string, dict)
+    :param listing_json: the contents of the listing document
+    :type  listing_json: string
+    :returns: a mapping from resource name to file path
+    :rtype: dict
     """
     def resource_name_to_filepath(name):
         return os.path.join(schema_dir, '{0}.json'.format(name))
 
-    listing, listing_json = _load_resource_listing(schema_dir)
-
-    return (
-        listing,
-        dict(
-            (resource, resource_name_to_filepath(resource))
-            for resource in find_resource_names(listing_json)
-        )
+    return dict(
+        (resource, resource_name_to_filepath(resource))
+        for resource in find_resource_names(listing_json)
     )
 
 
-def _load_resource_listing(schema_dir):
+def _load_resource_listing(resource_listing):
     """Load the resource listing from file, handling errors.
 
-    :param schema_dir: the directory schema files live inside
-    :type schema_dir: string
-    :returns: (resource listing filepath, resource listing json)
+    :param resource_listing: path to the api-docs resource listing file
+    :type  resource_listing: string
+    :returns: contents of the resource listing file
+    :rtype: dict
     """
-    resource_listing = os.path.join(schema_dir, API_DOCS_FILENAME)
     try:
         with open(resource_listing) as resource_listing_file:
-            resource_listing_json = simplejson.load(resource_listing_file)
+            return simplejson.load(resource_listing_file)
     # If not found, raise a more user-friendly error.
     except IOError:
         raise ResourceListingNotFoundError(
             'No resource listing found at {0}. Note that your json file '
             'must be named {1}'.format(resource_listing, API_DOCS_FILENAME)
         )
-    return resource_listing, resource_listing_json
 
 
-def compile_swagger_schema(schema_dir, should_validate_schemas):
+def compile_swagger_schema(schema_dir):
     """Build a SwaggerSchema from various files.
 
     :param schema_dir: the directory schema files live inside
     :type schema_dir: string
-    :param should_validate_schemas: if True, check schemas for correctness
-    :type should_validate_schemas: boolean
     :returns: a SwaggerSchema object
     """
-    listing, mapping = build_schema_mapping(schema_dir)
-    schema_resolvers = ingest_resources(
-        listing,
-        mapping,
-        schema_dir,
-        should_validate_schemas,
-    )
-    return SwaggerSchema(
-        listing,
-        mapping,
-        schema_resolvers,
-    )
+    listing_filename = os.path.join(schema_dir, API_DOCS_FILENAME)
+    listing_json = _load_resource_listing(listing_filename)
+    mapping = build_schema_mapping(schema_dir, listing_json)
```

Comment:
The reason I don't like that as much is because it makes the validation kind of implicit instead of being it's own call it's just a side-effect of building a schema.

Does ssv need `build_schema_mapping()` ? I was thinking that it would work off just `listing_filename`, which is why I left the duplicate lines.